corehq/apps/app_manager/suite_xml/post_process/workflow.py

```diff
@@ -77,54 +95,33 @@ def get_frame_children(self, command, target_module, module_only=False, include_
         :returns:   list of strings and DatumMeta objects. String represent stack commands
                     and DatumMeta's represent stack datums.
         """
-        match = re.search(r'^(m\d+)-(.*)', command)
-        if not match:
-            raise Exception("Unrecognized command: {}".format(command))
-        target_module_id = match.group(1)
-        target_form_id = match.group(2)
-
-        frame_children = []
+        module_command = id_strings.menu_id(module)
+        module_datums = self.get_module_datums(f'm{module.id}')
 
-        module_command = id_strings.menu_id(target_module)
-        module_datums = self.get_module_datums(target_module_id)
-        form_datums = module_datums[target_form_id]
         frame_children = []
-
         if module_command == id_strings.ROOT:
-            datums_list = self.root_module_datums
+            datums_list = self._root_module_datums
         else:
             datums_list = list(module_datums.values())  # [ [datums for f0], [datums for f1], ...]
-            root_module = target_module.root_module
-            if root_module and include_target_root:
+            root_module = module.root_module
+            if root_module and include_root_module:
                 datums_list = datums_list + list(self.get_module_datums(id_strings.menu_id(root_module)).values())
```

Comment:
I think this line is an issue (as @orangejenny identified [here](https://github.com/dimagi/commcare-hq/pull/25065/files#r313403004)).  The `commonprefix` call below this block removes all datum declarations that aren't a part of all forms in consideration.  However, this line appends all forms from the root module, so adding info from the root module actually _removes_ datums, which is kinda nonsensical.  I think `commonprefix` should be called once on the forms in the root module and once on the forms in this module.  Or something like that.

Regardless, I'm not going to change that in this PR, because it would be a behavior change for EoF navigation